core/src/main/scala/ai/h2o/sparkling/backend/api/ShutdownServlet.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ai.h2o.sparkling.backend.api
+
+import ai.h2o.sparkling.{H2OConf, H2OContext}
+import javax.servlet.Servlet
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+import water.exceptions.H2ONotFoundArgumentException
+
+/**
+  * Handler for the /3/shutdown POST command
+  */
+private[api] class ShutdownServlet extends ServletBase {
+
+  override def doPost(req: HttpServletRequest, resp: HttpServletResponse): Unit = {
+    val obj = req.getPathInfo match {
+      case null =>
+        H2OContext.get().foreach(_.stop(stopSparkContext = true))
+      case invalid => throw new H2ONotFoundArgumentException(s"Invalid endpoint $invalid")
+    }
+    sendResult(obj, resp)
+  }
+
+}
+
+object ShutdownServlet extends ServletRegister {
+
+  override protected def getEndpoints(): Array[String] = {
```

Comment:
Shouldn't `getEndpoints()`  return ip addresses?

Reply:
Nope, we return the request path. The `getEndpoints` method should be renamed